### PR TITLE
feat(SKE): support for DNS extension

### PR DESCRIPTION
**Pull request by Kumm-Kai**
This PR adds support for the new DNS extension.

[offical API spec](https://docs.api.eu01.stackit.cloud/documentation/ske/version/v1#tag/Cluster/operation/SkeService_CreateOrUpdateCluster)

I'm not sure if the acceptance tests will complete, as the cluster fails to get ready if the specified zones don't exist.

### CODE REVIEW

File: stackit/internal/services/ske/cluster/resource.go [hunk 1]
```diff
@@ -587,6 +601,21 @@ func (r *clusterResource) Schema(_ context.Context, _ resource.SchemaRequest, re
...
+								Description: "Flag to enable/disable DNS extensions",
+								Computed:    true,
+							},
+							"zones": schema.ListAttribute{
+								Description: "Specify a list of domain filters for externalDNS (e.g., `foo.runs.onstackit.cloud`)",
+								Computed:    true,
+								ElementType: types.StringType,
+							},
+						},
+					},
```
Comment (joaopalet) [inline stackit/internal/services/ske/cluster/resource.go:618]: Are these fields really just `Computed`? Shouldn't they be `Optional` to allow the user to set them?

File: stackit/internal/services/ske/ske_acc_test.go [hunk 2]
```diff
@@ -50,6 +50,8 @@ var clusterResource = map[string]string{
 	"extensions_acl_cidrs":                             "192.168.0.0/24",
 	"extensions_argus_enabled":                         "false",
 	"extensions_argus_instance_id":                     "aaaaaaaa-1111-2222-3333-444444444444", // A not-existing Argus ID let the creation time-out.
+	"extensions_dns_enabled":                           "true",
+	"extensions_dns_zones":                             "foo.onstackit.cloud",
```
Comment (joaopalet) [inline stackit/internal/services/ske/ske_acc_test.go:54]: We could also just create a simple DNS zone in the test and use it for the extension

File: stackit/internal/services/ske/cluster/resource.go (hunk 1 above)
Comment (Kumm-Kai) [reply stackit/internal/services/ske/cluster/resource.go:618]: Fixed

File: stackit/internal/services/ske/ske_acc_test.go (hunk 2 above)
Comment (Kumm-Kai) [reply stackit/internal/services/ske/ske_acc_test.go]: As discussed, we will use a dummy value for now.

**Review by joaopalet — APPROVED** [review]

File: stackit/internal/services/ske/cluster/datasource.go [hunk 3]
```diff
@@ -296,6 +296,21 @@ func (r *clusterDataSource) Schema(_ context.Context, _ datasource.SchemaRequest
 							},
 						},
 					},
+					"dns": schema.SingleNestedAttribute{
+						Description: "DNS extension configuration",
+						Computed:    true,
```
Comment (GokceGK) [inline stackit/internal/services/ske/cluster/datasource.go:301]:
```suggestion
						Optional:    true,
```

File: stackit/internal/services/ske/cluster/resource.go [hunk 4]
```diff
@@ -582,6 +596,21 @@ func (r *clusterResource) Schema(_ context.Context, _ resource.SchemaRequest, re
...
 							"allowed_cidrs": schema.ListAttribute{
 								Description: "Specify a list of CIDRs to whitelist.",
+								Required:    true,
+								ElementType: types.StringType,
+							},
+						},
+					},
+					"dns": schema.SingleNestedAttribute{
+						Description: "DNS extension configuration",
+						Computed:    true,
```
Comment (GokceGK) [inline stackit/internal/services/ske/cluster/resource.go:606]:
```suggestion
						Optional:    true,
```

File: stackit/internal/services/ske/ske_acc_test.go [hunk 5]
```diff
@@ -231,6 +239,9 @@ func TestAccSKE(t *testing.T) {
 					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extensions.acl.allowed_cidrs.0", clusterResource["extensions_acl_cidrs"]),
 					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extensions.argus.enabled", clusterResource["extensions_argus_enabled"]),
 					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extensions.argus.argus_instance_id", clusterResource["extensions_argus_instance_id"]),
+					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extenisons.dns.enabled", clusterResource["extensions_dns_enabled"]),
```
Comment (GokceGK) [inline stackit/internal/services/ske/ske_acc_test.go:242]:
```suggestion
					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extensions.dns.enabled", clusterResource["extensions_dns_enabled"]),
```

File: stackit/internal/services/ske/ske_acc_test.go [hunk 6]
```diff
@@ -231,6 +239,9 @@ func TestAccSKE(t *testing.T) {
...
+					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extenisons.dns.zones.#", "1"),
```
Comment (GokceGK) [inline stackit/internal/services/ske/ske_acc_test.go:243]:
```suggestion
					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extensions.dns.zones.#", "1"),
```

File: stackit/internal/services/ske/ske_acc_test.go [hunk 7]
```diff
@@ -231,6 +239,9 @@ func TestAccSKE(t *testing.T) {
...
+					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extenisons.dns.zones.0", clusterResource["extensions_dns_zones"]),
```
Comment (GokceGK) [inline stackit/internal/services/ske/ske_acc_test.go:244]:
```suggestion
					resource.TestCheckResourceAttr("stackit_ske_cluster.cluster", "extensions.dns.zones.0", clusterResource["extensions_dns_zones"]),
```

File: stackit/internal/services/ske/cluster/datasource.go (hunk 3 above)
Comment (Kumm-Kai) [reply stackit/internal/services/ske/cluster/datasource.go:301]: It's also `Computed` for ACL and argus. So what should it be?

Comment (GokceGK) [reply stackit/internal/services/ske/cluster/datasource.go:301]: oh yes, you are right. In datasource, it should be computed.

**Pull request merged**